datadog/resource_datadog_monitor.go

```diff
@@ -342,6 +349,22 @@ func resourceDatadogMonitorExists(d *schema.ResourceData, meta interface{}) (b b
 	return true, nil
 }
 
+func resourceDatadogMonitorCustomizeDiff(diff *schema.ResourceDiff, meta interface{}) error {
+	if _, ok := diff.GetOk("query"); !ok {
```

Comment:
Does this mean we only validate the monitor if the query changed? 
Will we have access to all the other non changed fields like thresholds?
(Can we debug log something here about not validating)

Reply:
No it means we only validate if we have a value, in particular when the query depends on another resource that's not resolved.

Reply:
Thanks. My main concern was that this `diff` object wouldn't contain the entire monitor object and only include what changed. But we discussed offline and that shouldn't be the case.